src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs

```diff
@@ -1878,6 +1887,36 @@ private SSHConnectionInfo()
             this.UserName = userName;
             this.ComputerName = computerName;
             this.KeyFilePath = keyFilePath;
+            this.Port = DefaultPort;
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="userName">User Name</param>
+        /// <param name="computerName">Computer Name</param>
+        /// <param name="keyFilePath">Key File Path</param>
+        /// <param name="port">Port number for connection (default 22)</param>
+        public SSHConnectionInfo(
+            string userName,
+            string computerName,
+            string keyFilePath,
+            int port)
```

Comment:
Please use the original constructor so as not to duplicate code:
public SSHConnectionInfo(string userName, string computerName, string keyFilePath, int port) : this(userName, computerName, keyFilePath)
{
    if ((port < MinPort) || (port > MaxPort))
    {
        ...
    }
    this.Port = (port != 0) port : DefaultPort;
}